### PR TITLE
Agent: add stylesheet for jetbrains

### PR DESCRIPTION
**Pull request by abeatrix**
Replace https://github.com/sourcegraph/cody/pull/4709

Initial mapping of the variables of JetBrains to VS code webview.

incremental improvements coming.

## Test plan

1. CODY_DIR=/Users/dpc/projects/cody ./gradlew :runIDE -PforceAgentBuild=true
2. Open Cody Experimental Chat UX sidebar, right click, Open DevTools
3. Check that a ton of JetBrains theme variables appear.
4. Check that the values parse OK
5. Switch theme in client and expect the webview updated accordingly

High contrast theme:

image

Drucula theme:

image

Light theme:

image

### CODE REVIEW

**Comment by dominiccooney** [comment]
What's the plan here, because you've dropped https://github.com/sourcegraph/cody/commit/95ceb538007527cd26c8d9055b4c56db48a41432 which you need to set the data-ide property, etc.

**Comment by abeatrix** [comment]
I tried updating the agent-blindings but getting:
```
+ /Users/bwork/dev/cody (122ms)
done /Users/bwork/dev/cody/index.scip
+ pnpm dlx ts-node agent/src/cli/scip-codegen/command.ts --output agent/bindings/kotlin/lib/src/main/kotlin/com/sourcegraph/cody/protocol_generated
.../Library/pnpm/store/v3/tmp/dlx-37738  |  +20 ++
.../Library/pnpm/store/v3/tmp/dlx-37738  | Progress: resolved 20, reused 20, downloaded 0, added 20, done
/Users/bwork/dev/cody/agent/src/cli/scip-codegen/SymbolTable.ts:48
            throw new Error(
                  ^
Error: no symbol: {
  "symbol": "scip-typescript npm cody-ai 1.24.0 src/chat/`protocol.ts`/indexedSignature2:",
  "debuggingInfo": []
}
    at SymbolTable.info (/Users/bwork/dev/cody/agent/src/cli/scip-codegen/SymbolTable.ts:48:19)
    at /Users/bwork/dev/cody/agent/src/cli/scip-codegen/JvmCodegen.ts:317:39
    at CodePrinter.block (/Users/bwork/dev/cody/vscode/src/completions/context/retrievers/tsc/CodePrinter.ts:45:9)
    at JvmCodegen.writeDataClass (/Users/bwork/dev/cody/agent/src/cli/scip-codegen/JvmCodegen.ts:300:11)
    at JvmCodegen.writeType (/Users/bwork/dev/cody/agent/src/cli/scip-codegen/JvmCodegen.ts:484:22)
    at JvmCodegen.run (/Users/bwork/dev/cody/agent/src/cli/scip-codegen/JvmCodegen.ts:60:22)
    at async Command.<anonymous> (/Users/bwork/dev/cody/agent/src/cli/scip-codegen/command.ts:80:9)
    at async Command.parseAsync (/Users/bwork/dev/cody/node_modules/.pnpm/commander@11.1.0/node_modules/commander/lib/command.js:936:5)
 ERROR  Command failed with exit code 1: ts-node agent/src/cli/scip-codegen/command.ts --output agent/bindings/kotlin/lib/src/main/kotlin/com/sourcegraph/cody/protocol_generated

pnpm: Command failed with exit code 1: ts-node agent/src/cli/scip-codegen/command.ts --output agent/bindings/kotlin/lib/src/main/kotlin/com/sourcegraph/cody/protocol_generated
    at makeError (/Users/bwork/.asdf/installs/pnpm/8.6.7/dist/pnpm.cjs:24796:17)
    at handlePromise (/Users/bwork/.asdf/installs/pnpm/8.6.7/dist/pnpm.cjs:25367:33)
    at process.processTicksAndRejections (node:internal/process/task_queues:95:5)
    at async Object.handler [as dlx] (/Users/bwork/.asdf/installs/pnpm/8.6.7/dist/pnpm.cjs:209900:7)
    at async /Users/bwork/.asdf/installs/pnpm/8.6.7/dist/pnpm.cjs:219307:21
    at async main (/Users/bwork/.asdf/installs/pnpm/8.6.7/dist/pnpm.cjs:219274:34)
    at async runPnpm (/Users/bwork/.asdf/installs/pnpm/8.6.7/dist/pnpm.cjs:219529:5)
    at async /Users/bwork/.asdf/installs/pnpm/8.6.7/dist/pnpm.cjs:219521:7
 ELIFECYCLE  Command failed with exit code 1.
 ```

File: vscode/webviews/promptEditor/nodes/ContextItemMentionNode.module.css [hunk 1]
```diff
@@ -40,3 +42,15 @@ body:is([data-vscode-theme-kind='vscode-high-contrast'], [data-vscode-theme-kind
         outline: solid 1px var(--vscode-input-foreground);
     }
 }
+
+/* color-mix is not supported on JetBrains webview */
+html:is([data-ide='JetBrains']) .context-item-mention-node {
```
Comment (dominiccooney) [inline vscode/webviews/promptEditor/nodes/ContextItemMentionNode.module.css:47]: Doesn't make sense to use `:is` with one selector, just write `html[data-ide=JetBrains]` and be done with it.

**Review by dominiccooney — APPROVED** [review]
This looks good, I see it has conflicts so please rebase etc.

Thank you for adding my 95ceb538007527cd26c8d9055b4c56db48a41432 back! Need this.

**Comment by dominiccooney** [comment]
Thanks for working on this!

Bit of commit message polish would be great: instead of WIP we could say something about incremental improvements coming.

Do you want to land this to keep the diff small and do anything needed for font sizes, sizes, etc. in a follow up?

**Comment by abeatrix** [comment]
> Thanks for working on this!
> 
> Bit of commit message polish would be great: instead of WIP we could say something about incremental improvements coming.
> 
> Do you want to land this to keep the diff small and do anything needed for font sizes, sizes, etc. in a follow up?

Yea let's do this! Let me update the PR with your feedback and rebase it before I merge, thanks for reviewing!

**Pull request merged**